allennlp/modules/token_embedders/pretrained_transformer_embedder.py

```diff
@@ -16,12 +16,17 @@ def __init__(self, model_name: str) -> None:
         self.transformer_model = AutoModel.from_pretrained(model_name)
         # I'm not sure if this works for all models; open an issue on github if you find a case
         # where it doesn't work.
+        self.model_type = model_name.split("-")[0]
         self.output_dim = self.transformer_model.config.hidden_size
 
     @overrides
     def get_output_dim(self):
         return self.output_dim
 
-    def forward(self, token_ids: torch.LongTensor) -> torch.Tensor:  # type: ignore
+    def forward(
+        self, token_ids: torch.LongTensor, attention_mask: torch.LongTensor
+    ) -> torch.Tensor:  # type: ignore
+        if self.model_type == "xlnet":
+            return self.transformer_model(input_ids=token_ids, input_mask=attention_mask)[0]
```

Comment:
Looking at the code, I'm pretty sure this will do the wrong thing - if you want to use `input_mask`, you need to compute the inverse of what you computed above.  We should just remove this case.

Reply:
Thank you for your advice.
I read huggingface's xlnet implementation and seem to have misunderstood. I will delete this case.